File: android/app/src/main/java/com/adqmobile/socialnetwork/domain/user/usecases/LoginUseCase.kt
package com.adqmobile.socialnetwork.domain.user.usecases

import com.adqmobile.socialnetwork.domain.user.Login
import com.adqmobile.socialnetwork.domain.user.User
import com.adqmobile.socialnetwork.domain.user.UserRepository

class LoginUseCase(private val userRepository: UserRepository,
                   private val storeUserUseCase: StoreUserUseCase) {

    suspend fun run(login: Login): User {
        val user = userRepository.login(login)
        storeUserUseCase.run(user)
        return user
    }
}

File: web/src/domain/post/usecases/LoginUseCase.ts
import UserRepository from "../../user/UserRepository";
import User from "../../user/User";
import Login from "../../user/Login";

export default class LoginUseCase {
    userRepository: UserRepository;

    constructor(userRepository: UserRepository) {
        this.userRepository = userRepository;
    }

    async run(login: Login): Promise<User> {
        return await this.userRepository.login(login);
    }
}